src/main/java/org/apache/cassandra/sidecar/restore/RestoreProcessor.java

```diff
@@ -151,7 +152,8 @@ public void execute(Promise<Void> promise)
                                                          restoreJobUtil,
                                                          localTokenRangesProvider,
                                                          metrics);
-            activeTasks.add(task);
+
+            activeTasks.put(task, slowTaskThresholdInSeconds);
             pool.executeBlocking(task, false) // unordered; run in parallel
             .onSuccess(restoreSlice -> {
```

Comment:
do we need to remove the task from `activeTasks` when a failure occurs?

Reply:
task is removed in the `onComplete` block, regardless of success or failure.